### PR TITLE
Updated AppManifests

### PR DESCRIPTION
**Pull request by adithyaakrishna**
Signed-off-by: Adithya Krishna <aadithya794@gmail.com>

#### Summary
Updates AppManifests according to https://github.com/mattermost/mattermost-plugin-apps/blob/3518d938db7271362d55b53fda492facf0006210/apps/manifest.go#L22

#### Ticket Link
This PR is Related to https://github.com/mattermost/mattermost-app-zendesk/issues/78

### CODE REVIEW

File: src/types/apps.ts [hunk 1]
```diff
@@ -5,12 +5,28 @@
...
+  description?: string;
+  icon?: string;
+  bindings?: AppBinding;
+  on_install?: AppBinding;
+  on_version_changed?: AppBinding;
+  on_uninstall?: AppBinding;
+  on_disable?: AppBinding;
+  on_enable?: AppBinding;
+  get_oauth2_connect_url?: AppBinding;
+  on_oauth2_complete?: AppBinding;
```
Comment (adithyaakrishna) [inline src/types/apps.ts:22]: I wasn't sure completely about what data type to have for this, as the go file mentioned *Call I've added AppBinding here. Please do lmk if I need to change it to something else :)

Comment (jfrerich) [reply src/types/apps.ts:22]: The `Call` struct can be found here. https://github.com/adithyaakrishna/mattermost-redux/blob/master/src/types/apps.ts#L79-L83

Comment (jfrerich) [reply src/types/apps.ts:22]: The `kubeless_functions` and `aws_lambda` types are not yet defined in this file.  I'm ok with commenting those out if you would like or you can implement them to match the following go structs

https://github.com/mattermost/mattermost-plugin-apps/blob/3518d938db7271362d55b53fda492facf0006210/apps/manifest.go#L116

https://github.com/mattermost/mattermost-plugin-apps/blob/3518d938db7271362d55b53fda492facf0006210/apps/manifest.go#L178

**Review by jfrerich — CHANGES_REQUESTED** [review]
Thanks for the PR, @adithyaakrishna! 

I've left some notes for you. If it takes a while for this PR to get merged, I will change the issue to only create the Manifest Type in Redux and create a separate issue for importing into the Zendesk app.

**Comment by adithyaakrishna** [comment]
Okay sure, I'll add the kubeless and aws types here as well :)

File: src/types/apps.ts (hunk 1 above)
Comment (mickmister) [reply src/types/apps.ts:22]: @levb Are we at a stable enough spot with the manifest structure where we should be updating the types in this repo? I suppose it doesn't hurt to keep them in sync in general

File: src/types/apps.ts [hunk 2]
```diff
@@ -5,12 +5,28 @@
...
+  requested_permissions?: string[];
+  requested_locations?: string[];
+  root_url?: string;
+  aws_lambda?: string;
+  kubeless_functions?: string;
```
Comment (adithyaakrishna) [inline src/types/apps.ts:27]: My Notes: Comment Them

**Comment by hmhealey** [comment]
Thanks for the PR, @adithyaakrishna. Unfortunately, I think we might have run into some confusion over where this change should be happening.

@jfrerich We haven't been shipping mattermost-redux since it was moved out of this repo and into the web app. We could update it in the web app's version of mattermost-redux, but it won't be available to Apps wanting to use this type either way. Maybe we could look at shipping those types to Apps some other way since we're not in a state to ship mattermost-redux as a whole?

File: src/types/apps.ts (hunk 1 above)
Comment (hmhealey) [reply src/types/apps.ts:22]: See my comment below. Even if the manifest structure is stable, we haven't been shipping mattermost-redux, so these new types won't be usable for any Apps wanting to import them. I'm happy to discuss what we might do here though

**Comment by adithyaakrishna** [comment]
@hmhealey Gotcha, should I move this to the other Repo? 👀 
That is to this file, https://github.com/mattermost/mattermost-webapp/blob/master/packages/mattermost-redux/src/types/apps.ts ?

**Comment by hmhealey** [comment]
Yeah, that'd probably make sense, but that won't make them accessible for use in mattermost-app-zendesk, so we might want to wait to see what @jfrerich was intending with the ticket.

**Comment by mickmister** [comment]
@hmhealey @jfrerich
As I mentioned [here](https://community-daily.mattermost.com/core/pl/xw5c6tw1s7n1tgcuabdk1utqbw), I'm proposing we move server-side relevant pieces of `mattermost-redux` to a separate repo. Either way we definitely need a way to continue updating types for Node.js applications going forward.

**Comment by adithyaakrishna** [comment]
> @hmhealey @jfrerich As I mentioned [here](https://community-daily.mattermost.com/core/pl/xw5c6tw1s7n1tgcuabdk1utqbw), I'm proposing we move server-side relevant pieces of `mattermost-redux` to a separate repo. Either way we definitely need a way to continue updating types for Node.js applications going forward.

Gotcha @mickmister 💯 
Will close this and move it there :)

**Comment by hmhealey** [comment]
Closing this in favour of the mattermost-webapp PR

**Pull request closed without merging**